pkg/controller/command/command_config_maps.go

```diff
@@ -49,16 +53,31 @@ func (c *configMaps) ensureCommandConfigExist(postgresAddress, ConfigEndpoint st
 }
 
 func (c *configMaps) ensureCommandInitConfigExist(webUIPort, swiftProxyPort, keystonePort int, webUIAddress, swiftProxyAddress, keystoneAddress, keystoneAuthProtocol, postgresAddress, ConfigEndpoint string, podIP string) error {
+	configAPIURL := fmt.Sprintf("https://%v:%v", ConfigEndpoint, contrail.ConfigApiPort)
+	telemetryURL := fmt.Sprintf("https://%v:%v", ConfigEndpoint, contrail.AnalyticsApiPort)
+	webUIURL := fmt.Sprintf("https://%v:%v", webUIAddress, webUIPort)
+	keystoneURL := fmt.Sprintf("%v://%v:%v", keystoneAuthProtocol, keystoneAddress, keystonePort)
+	swiftProxyURL := fmt.Sprintf("https://%v:%v", swiftProxyAddress, swiftProxyPort)
+
+	ces := []contrail.CommandEndpoint{
+		{Name: "nodejs", PrivateURL: webUIURL, PublicURL: webUIURL},
+		{Name: "telemetry", PrivateURL: telemetryURL, PublicURL: telemetryURL},
+		{Name: "config", PrivateURL: configAPIURL, PublicURL: configAPIURL},
+		{Name: "keystone", PrivateURL: keystoneURL, PublicURL: keystoneURL},
+		{Name: "swift", PrivateURL: swiftProxyURL, PublicURL: swiftProxyURL},
+	}
+	ces = append(ces, c.ccSpec.ServiceConfiguration.Endpoints...)
+	bes := []bootstrapEndpoint{}
+	for _, e := range ces {
+		bes = append(bes, *newBootstrapEndpoint(e))
```

Comment:
do we need to return an pointer from newBootstrapEndpoint method if you are just using objects and then you append the object to the slice

Reply:
changed